### DISCUSSION TITLE
Re-usable Responses & blocks of data across multiple APIs

### DISCUSSION THREAD

**Discussion by ghost** [ApicurioArchive/apicurio-studio#2099] [Q&A] [answered]
Hi,

I am exploring **apicurio studio** and I was wondering if its possible to refer to reusable blocks of data across APIs.
I see the option to use reusable data blocks defined in the current API but I was not able to see how could be done across multiple files?

If yes, how?

> **ex:** if we have API error responses standardised across all our APIs, we would create those are reusable data "somewhere" and then refer to them on all our APIs

many thanks in advance 
. Miguel

**Comment by EricWittmann** [comment, marked as answer]
Hi @miguel-pinto-2 - sorry for the late response, I've been out for awhile.  :(

Yes, this can be done in Apicurio Studio.  You obviously must first create an API that contains the data types or responses you would like to re-use across multiple APIs.  Once you have done that, then you can *import* some or all of those types.  To do that, click the little "box with an arrow" icon in the Data Types or Responses section of the left nav:

![image](https://user-images.githubusercontent.com/1890703/222140452-d77a0ae0-3774-4ad6-a24c-e1283d3a4955.png)

That will bring up a little wizard you can use to import what you want to use:

![image](https://user-images.githubusercontent.com/1890703/222140554-9660782a-ced7-45f8-ba30-d67b430eee44.png)

Once you have successfully imported a data type or response, they will be marked as such in the list:

![image](https://user-images.githubusercontent.com/1890703/222140803-2e746f6f-23b7-4e36-bf63-beadd331e9e2.png)
